apartment_2021_project_yinon_chen/main.c

```diff
@@ -170,76 +203,133 @@ void commandHandler(char* inputLine, apartmentList* aptList){
         printf("short_history init");
 }
 
-apartmentList makeEmptyList()
+
+void historyHandler(char *inputLine, int index, char *short_term_history[N], HistoryList *historyList) {
+    char fullCommandLine[strlen(inputLine)];
+    strcpy(fullCommandLine, inputLine);
+
+    if (index >= 6)//check if the array is full
+    {
+        insertToArchive(historyList,short_term_history[6]);
+
+    }
+    //if not full start to push string to th
+
+    for (int i = index; i >= 1; i--) {
+        short_term_history[i + 1] = short_term_history[i];
+    }
+
+
+    short_term_history[0] = fullCommandLine;
```

Comment:
I think (maybe I am wrong) fullCommandLine is freed at the end of the scope (this function). Therefore, we will lose the string. If so, we do need to allocate memory in line _char fullCommandLine[strlen(inputLine)];_ so other scopes can use it